packages/kbn-apm-synthtrace/src/cli/utils/synthtrace_worker.ts

```diff
@@ -65,8 +75,12 @@ async function start() {
   }, 5000);
 
   await logger.perf('index_scenario', async () => {
-    await apmEsClient.index(generators);
-    await apmEsClient.refresh();
+    const promises = generatorsAndClientsArray.map(async ({ client, generator }) => {
+      await client.index(generator);
+      await client.refresh();
+    });
```

Comment:
can you call index and refresh once per client?

Reply:
Are you asking or telling? This would indeed call index and refresh once per client.

Reply:
no it calls it once per scenario + client. I am curious if we can just have one stream (per process) and index data from it, like we had before.

Reply:
For anyone else coming to this conversation, this has been resolved now, the comment was applicable for Live Data Upload and not Historical